### ISSUE TITLE
Exclude own posts from bookmarks list

### ISSUE THREAD

**Issue by realpixelcode** [sk22/megalodon#855]
**Is your feature request related to a problem? Please describe.**

Mastodon has a feature to auto-delete old posts. You can make exceptions for bookmarked posts, but that clutters your bookmarks list with your own posts when looking for someone else's post in your bookmarks.

**Describe the solution you'd like**

There should be a check mark in the bookmarks to include or exclude your own posts in the list.

**Describe alternatives you've considered**

None.

**Additional context**

![grafik](https://github.com/sk22/megalodon/assets/52963327/6504dd6b-2b43-4009-9f65-bd9c9a14299d)

![grafik](https://github.com/sk22/megalodon/assets/52963327/9be5196f-a8b2-475a-9562-42fdf4579838)

**Comment by FineFindus** [comment]
I don't think the proposed solution is particularly elegant, the use cases of it are quite limited. I think it is better to solve the underlying problem of not being able to find a post easily. A better solution would be to add a search function to the bookmarks list (and possibly others as well), but I think both require server-side implementation.

**Comment by realpixelcode** [comment]
Yes, as far as I know, bookmarks are not fetched all at once but lazily, which makes searching them impossible for the client on its own.

However, (not) displaying fetched posts that fulfill certain criteria should technically be perfectly fine, since the filtering can be done individually per post.

I don't think a simple filtering button would be too unelegant, considering that (in the official app) there are already three filtering buttons present in the profile timeline (posts, posts and replies, media). In this sense, adding another button of this kind wouldn't really violate the general “UI ecosystem” within the app.

While I agree that a search function wouldn't be bad either, it would solve a slightly different problem: finding a specific post instead of unspecifically browsing posts that fulfill certain criteria.
